recipes/seadex-essentials/all/conanfile.py

```diff
@@ -0,0 +1,102 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.files import export_conandata_patches, get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+class SeadexEssentialsConan(ConanFile):
+    name = "seadex-essentials"
+    description = "essentials is a small c++ library that offers very basic capabilities for applications and libraries."
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://essentials.seadex.de/"
+    topics = ("utility", "c++")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "spdlog/*:header_only": True
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "8.3",
+            "clang": "12",
+            "Visual Studio": "16",
+            "msvc": "192",
+            "apple-clang": "10"
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.shared = False
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("spdlog/1.11.0", headers=True, transitive_headers=True)
+        self.requires("fmt/9.1.0", headers=True, transitive_headers=True)
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "Windows", "Macos"]:
+            raise ConanInvalidConfiguration("This recipe supports only Linux, Windows and macOS!")
```

Comment:
Aren't there more than those 3 supported by Conan? I can see in the settings.yml file that there are "WindowsStore", "watchOS", "tvOS", "Android", etc...

Reply:
Yes but ConanCenter has a far more limited scope, so the expectation is that it should work for these three however the majority of the rest do not work for the majority of recipes

Reply:
I understand. But isn't it better to let the user of the library from conan-center know the platforms are not supported and not let him search for error fixing instead?

Reply:
We would need to add this to to 1500 recipes which does not work either. Plus even if it's not tested it pretty often they actually work. Conan pass in a lot of the correct things to CMake when doing Android or iOS and in general it's better to not artificially block them.

If there are issues reporting a platform is broken upstream, please provide a link so people can contribute. Otherwise we should can take the "less is more" approach

Reply:
Removed the check.